File: Ahorn/entities/SturdyFakeWall.jl
module Sardine7SturdyFakeWall

using ..Ahorn, Maple

@mapdef Entity "Sardine7/SturdyFakeWall" SturdyFakeWall(x::Integer, y::Integer, playTransitionReveal::Bool=false, playNormalReveal::Bool=true, permanent::Bool=true, blendin::Bool=true, revealSound::String="event:/game/general/secret_revealed", tiletype::String="3")

const placements = Ahorn.PlacementDict(
    "Sturdy Fake Wall (Sardine7)" => Ahorn.EntityPlacement(
        SturdyFakeWall,
        "rectangle",
        Dict{String, Any}(),
        Ahorn.tileEntityFinalizer
    ),
)

Ahorn.editingOptions(entity::SturdyFakeWall) = Dict{String, Any}(
    "tiletype" => Ahorn.tiletypeEditingOptions()
)

Ahorn.minimumSize(entity::SturdyFakeWall) = 8, 8
Ahorn.resizable(entity::SturdyFakeWall) = true, true

Ahorn.selection(entity::SturdyFakeWall) = Ahorn.getEntityRectangle(entity)

Ahorn.renderAbs(ctx::Ahorn.Cairo.CairoContext, entity::SturdyFakeWall, room::Maple.Room) = Ahorn.drawTileEntity(ctx, room, entity, alpha=0.7)

end

File: Loenn/entities/sturdy_fake_wall.lua
local fakeTilesHelper = require("helpers.fake_tiles")

local sturdyFakeWall = {}

sturdyFakeWall.name = "Sardine7/SturdyFakeWall"
sturdyFakeWall.depth = -13000

function sturdyFakeWall.placements()
    return {
        name = "sturdy_fake_wall",
        data = {
            tiletype = "3",
            width = 8,
            height = 8,
            playTransitionReveal = false,
            playNormalReveal = true,
            permanent = true,
            blendin = true,
            revealSound = "event:/game/general/secret_revealed"
        }
    }
end

sturdyFakeWall.sprite = fakeTilesHelper.getEntitySpriteFunction("tiletype", true, "tilesFg", {1.0, 1.0, 1.0, 0.7})
sturdyFakeWall.fieldInformation = fakeTilesHelper.getFieldInformation("tiletype")

return sturdyFakeWall
